src/core/ext/transport/chaotic_good/client_transport.h

```diff
@@ -55,56 +70,124 @@ class ClientTransport {
       writer_.reset();
     }
+    if (reader_ != nullptr) {
+      reader_.reset();
+    }
   }
   auto AddStream(CallArgs call_args) {
     // At this point, the connection is set up.
     // Start sending data frames.
     uint64_t stream_id;
+    InterActivityPipe<ServerFrame, server_frame_queue_size_> server_frames;
     {
       MutexLock lock(&mu_);
       stream_id = next_stream_id_++;
+      stream_map_.insert(
+          std::pair<uint32_t,
+                    std::shared_ptr<InterActivityPipe<
+                        ServerFrame, server_frame_queue_size_>::Sender>>(
+              stream_id, std::make_shared<InterActivityPipe<
+                             ServerFrame, server_frame_queue_size_>::Sender>(
+                             std::move(server_frames.sender))));
     }
-    return Seq(
-        // Continuously send data frame with client to server messages.
-        ForEach(std::move(*call_args.client_to_server_messages),
-                [stream_id, initial_frame = true,
-                 client_initial_metadata =
-                     std::move(call_args.client_initial_metadata),
-                 outgoing_frames = outgoing_frames_.MakeSender()](
-                    MessageHandle result) mutable {
-                  ClientFragmentFrame frame;
-                  frame.stream_id = stream_id;
-                  frame.message = std::move(result);
-                  if (initial_frame) {
-                    // Send initial frame with client intial metadata.
-                    frame.headers = std::move(client_initial_metadata);
-                    initial_frame = false;
-                  }
-                  return Seq(
-                      outgoing_frames.Send(ClientFrame(std::move(frame))),
-                      [](bool success) -> absl::Status {
-                        if (!success) {
-                          return absl::InternalError(
-                              "Send frame to outgoing_frames failed.");
-                        }
-                        return absl::OkStatus();
-                      });
-                }));
+    return TrySeq(
+        TryJoin(
+            // Continuously send client frame with client to server messages.
+            ForEach(std::move(*call_args.client_to_server_messages),
+                    [stream_id, initial_frame = true,
+                     client_initial_metadata =
+                         std::move(call_args.client_initial_metadata),
+                     outgoing_frames = outgoing_frames_.MakeSender()](
+                        MessageHandle result) mutable {
+                      ClientFragmentFrame frame;
+                      frame.stream_id = stream_id;
+                      frame.message = std::move(result);
+                      if (initial_frame) {
+                        // Send initial frame with client intial metadata.
+                        frame.headers = std::move(client_initial_metadata);
+                        initial_frame = false;
```

Comment:
The `initial_frame` will be false after the first frame is sent because the lambda function is mutable and the capture list value could be changed in runtime (I double checked this with printing out`std::cout<<initial_frame` in the lambda function).

Reply:
Thanks for the clarification.